Cloth.Application/Interfaces/IHttpServiceClient.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Cloth.Application.Interfaces;
+using Cloth.Domain.Entities;
+
+public interface IHttpServiceClient
```

Comment:
Interface and implementation the names should e couple to the external api (www.mocky.io).
This means, your name now is too generic, but in the other hands contains a method GetClothsAsync();

It is accepted to put a prefix on the domain of the service that you access via HTTP request, in order to comply with this good practice you need to rename it. Because as you can see, you yourself wrote a method that does not have a generic name, but describes what it will take. And the service name will tell you from where.
I**Mocky**HttpServiceClient.cs / **Mocky**HttpServiceClient.cs

If you access any other external APIs, their names will be:
IDigitallHttpClient, IGoogleAnalyticsHttpClient, IAwsHttpClient and etc. 

BUT:
If you want to keep it that way with one generic implementation for everyone.
Then you have to refactor it so that the whole implementation works with everyone, the request/response model and the methods should also have generic names instead of GetCloth it will be GetAsync
and when it becomes so generic, it will have to be assembled in its own assembly, because it will be common assembly.